library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -438,119 +261,110 @@ public void endWindow()
     currentWindowRecoveryState = new MutablePair<>();
   }
 
-  public int getPartitionCount()
-  {
-    return partitionCount;
-  }
-
-  public void setPartitionCount(int partitionCount)
+  @Override
+  public void deactivate()
   {
-    this.partitionCount = partitionCount;
+    scanService.shutdownNow();
+    store.disconnect();
   }
 
-  @Override
-  public void activate(Context cntxt)
+  protected void pollRecords()
   {
-    if (context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID
-        && context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) < windowManager.getLargestRecoveryWindow()) {
-      // If it is a replay state, don't start any threads here
+    if (isPolled) {
       return;
     }
-  }
-
-  @Override
-  public void deactivate()
-  {
     try {
-      if (dbPoller != null && dbPoller.isAlive()) {
-        dbPoller.interrupt();
-        dbPoller.join();
+      ps.setFetchSize(getFetchSize());
+      ResultSet result = ps.executeQuery();
+      if (result.next()) {
+        do {
+          emitQueue.add(getTuple(result));
+        } while (result.next());
       }
-    } catch (InterruptedException ex) {
-      // log and ignore, ending execution anyway
-      LOG.error("exception in poller thread: ", ex);
+      isPolled = true;
```

Comment:
the fetchSize, is provided to the PreparedStatement to tell it how many records to fetch at a time i..e. we can say how many records to fetch as a page. Multiple of such pages will be fetched based on how many records are present. e.g. we have 1000 records and fetchSize is 100, then 10 pages of 100 records each will be fetched.